library/ssl_tls13_server.c

```diff
@@ -2949,6 +2957,49 @@ static int ssl_tls13_prepare_new_session_ticket(mbedtls_ssl_context *ssl,
     return 0;
 }
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+/* RFC 8446 section 4.2.10
+ *
+ * struct {
+ *   select ( Handshake.msg_type ) {
+ *     case new_session_ticket:   uint32 max_early_data_size;
+ *     ...
+ *   };
+ * } EarlyDataIndication;
+ */
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int ssl_tls13_write_early_data_ext_of_nst(mbedtls_ssl_context *ssl,
+                                                 unsigned char *buf,
+                                                 const unsigned char *end,
+                                                 size_t *out_len)
+{
+    unsigned char *p = buf;
+    *out_len = 0;
+
+    if ((ssl->session->ticket_flags &
+         MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_EARLY_DATA) == 0) {
```

Comment:
How about adding new helper `mbedtls_ssl_session_ticket_allow_early_data` and use it everywhere?
```c
static inline unsigned int mbedtls_ssl_session_ticket_allow_early_data(
    mbedtls_ssl_session *session)
{
    return !mbedtls_ssl_session_check_ticket_flags(session,
                                                   MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_EARLY_DATA);
}
```

Note: This might have conflict with #8414.